pkg/ebpf/telemetry/modifier.go

```diff
@@ -135,3 +150,24 @@ func (t *ErrorsTelemetryModifier) AfterInit(m *manager.Manager, module names.Mod
 
 	return nil
 }
+
+// BeforeStop stops the perf collector from telemetry and removes the modules from the telemetry maps.
+func (t *ErrorsTelemetryModifier) BeforeStop(m *manager.Manager, module names.ModuleName) error {
+	if perfCollector != nil {
+		perfCollector.unregisterTelemetry(m)
+	}
```

Comment:
do we really want to do it in the modifer BeforeStop? that feels very wrong design wise.
This should be handled by the `ebpfErrorsTelemetry` itself and not by the modifier

Reply:
`perfCollector` is not even part of errors telemetry. This should definitely not be done here.